src/linker/Linker/MessageContainer.cs

```diff
@@ -132,6 +132,16 @@ private static MessageContainer CreateWarningMessageContainer (LinkContext conte
 			if (version > context.WarnVersion)
 				return Empty;
 
+			if (origin.MemberDefinition != null) {
+				var assembly = origin.MemberDefinition.DeclaringType.Module.Assembly;
+				var assemblyName = assembly.Name.Name;
+				if (context.AreWarningsCollapsed (assemblyName)) {
```

Comment:
Shouldn't this be the first check?

Reply:
Not sure what you mean exactly, but https://github.com/mono/linker/pull/1937#discussion_r607275309 may address it. This is only trying to collapse the trim analysis warnings (which we should be able to "blame" on a particular assembly).